application/models/Permission.php

```diff
@@ -575,7 +575,7 @@ public function hasPermission($iEntityID, $sEntityName, $sPermission, $sCRUD = '
 
         /* Always return true if you are the owner : this can be done in core plugin ? */
         // TODO: give the rights to owner adding line in permissions table, so it will return true with the normal way
-        if ($iUserID == $this->getEntityOwnerId($iEntityID, $sEntityName)) {
+        if ($iUserID && $iUserID == $this->getEntityOwnerId($iEntityID, $sEntityName)) {
```

Comment:
Can you be more clear what you test with user id? empty? null? the string "0"?

Reply:
https://github.com/LimeSurvey/LimeSurvey/pull/1755/commits/5838c394430bcc6093f21eae98eb204c9aad5c1d#diff-10077a3f8791500882779e35a827caf5e6578d5140625094d224de5e3c166780R527